datascience_eda/datascience_eda.py

```diff
@@ -74,31 +168,293 @@ def explore_clustering(
 
-    # add all plots to result dictionary
+    Parameters
+    ----------
+    df : pandas.DataFrame
+        the dataset, should be transformed with StandardScaler
+    num_cols : list, optional
+        list of numeric column names, in case of None, get all numeric columns
+    metric : str, optional
+        metric, by default "euclidean"
+    n_clusters : list, optional
+        list of n_clusters hyperparams, by default range(2, 9)
+    include_silhouette : bool, optional
+        whether Silhouette plots should be generated, by default True
+    include_PCA : bool, optional
+        whether PCA plots should be generated, by default True
 
-    # ---clustering with DBSCAN---------
-    # get eps and min_samples from hyperparameter_dict["DBSCAN"]
+    Returns
+    -------
+    dict
+        a dictionary with key=type of plot, value=list of plots
+
+    Examples
+    -------
+    >>> original_df = pd.read_csv(currentdir + "/data/menu.csv")
+    >>> numeric_features = eda.get_numeric_columns(original_df)
+    >>> drop_features = []
+    >>> numeric_transformer = make_pipeline(SimpleImputer(), StandardScaler())
+    >>> preprocessor = make_column_transformer(
+    >>>     (numeric_transformer, numeric_features), ("drop", drop_features)
+    >>> )
+    >>> df = pd.DataFrame(
+    >>>     data=preprocessor.fit_transform(original_df), columns=numeric_features
+    >>> )
+    >>> explore_KMeans_clusterting(df)
+    """
+    if num_cols is None:
+        num_cols = get_numeric_columns(df)
+    else:
+        _verify_numeric_cols(df, num_cols)
+    x = df[num_cols]
+    results = {}
+    if 1 in n_clusters:
+        raise Exception("n_cluster cannot be 1")
+
+    print("------------------------")
+    print("K-MEANS CLUSTERING")
+    print("------------------------")
+
+    if len(n_clusters) > 1:
+        print("Generating KElbow plot for KMeans.")
+        # visualize using KElbowVisualizer
+        kmeans = KMeans()
+        elbow_visualizer = KElbowVisualizer(kmeans, k=n_clusters)
+        elbow_visualizer.fit(x)  # Fit the data to the visualizer
+        elbow_visualizer.show()
+        plt.close()
+        elbow_visualizer.k = elbow_visualizer.elbow_value_  # fix printing issue
+        results["KElbow"] = elbow_visualizer
+    else:
+        results["KElbow"] = None
 
     # visualize using SilhouetteVisualizer
+    print("Generating Silhouette & PCA plots")
+    silhouette_plots = []
+    pca_plots = []
+    for k in n_clusters:
+        print(f"Number of clusters: {k}")
+
+        kmeans = KMeans(k)
+
+        if include_silhouette:
+            s_visualizer = SilhouetteVisualizer(kmeans, colors="yellowbrick")
+            s_visualizer.fit(x)  # Fit the data to the visualizer
+            s_visualizer.show()
+            plt.clf()
+            plt.close()
+            silhouette_plots.append(s_visualizer)
+        else:
+            silhouette_plots.append(None)
+
+        # PCA plots
+        if include_silhouette:
```

Comment:
Could you please check if this if include_silhouette is if include_pca?
```suggestion
        if include_pca:
```